### PR TITLE
added documentation for fedrec/worker_dataset.py

### PR DESCRIPTION
**Pull request by bashirk**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Relevant Issue
Issue #Tag the issue here

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/EnvisEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/EnvisEdge/blob/main/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/NimbleEdge/EnvisEdge/labels)
- [ ] My changes are covered by tests

### CODE REVIEW

File: fedrec/utilities/random_state.py [hunk 1]
```diff
@@ -17,6 +42,7 @@ def __init__(self):
         ]
 
     def restore(self):
+        """This class method sets and restores the state for Numpy, PyTorch, & RNGs."""
```
Comment (divyanshugit) [inline fedrec/utilities/random_state.py:45]: You can remove this as you already mentioned it.

File: fedrec/utilities/random_state.py [hunk 2]
```diff
@@ -25,13 +51,43 @@ def restore(self):
...
+    _active: bool
+        set the active state of the RandomContext to False
+
+    Method
+    -------
+    restore()
+        sets and restores the state for Numpy, Torch, & Torch RNGs
+    
+    """
+    """Save and restore state of PyTorch, NumPy, Python RNGs."""
```
Comment (divyanshugit) [inline fedrec/utilities/random_state.py:82]: You can remove this as well and the comment on line 45 as well

File: fedrec/utilities/random_state.py [hunk 3]
```diff
@@ -7,7 +7,32 @@
 
 
 class RandomState:
+    """
+    This class represents the random state, and includes a method for restoring states.
```
Comment (divyanshugit) [inline fedrec/utilities/random_state.py:11]: The repetition of `This class` doesn't look good. You can re-write it as `It` or `The RandomState`. Please do it for all

File: fedrec/utilities/random_state.py [hunk 4]
```diff
@@ -7,7 +7,32 @@
...
+
+    Method
+    -------
+    restore()
+        sets and restores the state for Numpy, Torch, & Torch RNGs
+    
+    """
+
     def __init__(self):
+        """Gets the state for Numpy, PyTorch, & RNGs."""
```
Comment (divyanshugit) [inline fedrec/utilities/random_state.py:35]: You can remove this as well

**Review by divyanshugit — CHANGES_REQUESTED** [review]

**Comment by divyanshugit** [comment]
Please refrain from re-writing the same line again and again in the file. And please add a proper description for the PR

Thanks

File: fedrec/utilities/registry.py [hunk 5]
```diff
@@ -23,31 +23,14 @@
...
-          kind of object (eg. model, trainer).
-    name: str
-          Sub-key under kind key, used to specify name of
-          of the object definition.
-
-    Returns
-    ----------
-    callable:
-          Decorator function to store object definition.
-
```
[the remark this answers is not in the document]
Comment (bashirk) [reply fedrec/utilities/registry.py:43]: Yes, it's because I wanted to rewrite the comments.

File: fedrec/utilities/random_state.py (hunk 1 above)
Comment (bashirk) [reply fedrec/utilities/random_state.py:45]: Alright. Thanks.

File: fedrec/utilities/random_state.py (hunk 3 above)
Comment (bashirk) [reply fedrec/utilities/random_state.py:11]: Thanks, I'll update accordingly.

**Pull request closed without merging**